### PR TITLE
Bump SparseArrays dependencies for julia 1.9

### LINKED ISSUE #2939: LS fails for Julia 1.9
With Julia latest , I get this error. The `v1` directory it mentions has drwxr-x---. I can't send lines from the editor to the repl. it always notifies with "Connection to server got closed. Server will not be restarted." This only happens when I've set the active binary to julia latest. It works fine with Julia 1.7.

```
(null) pkg> activate --temp
  Activating new project at `/tmp/jl_fWQO0H`

julia> using JuliaFormatter

julia> VERSION
v"1.9.0-DEV.850"
```

```
┌ Error: Some Julia code in the VS Code extension crashed
└ @ Main ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/error_handler.jl:15
ERROR: Failed to precompile LanguageServer [2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7] to /home/u/.config/Code/User/globalStorage/julialang.language-julia/lsdepot/v1/compiled/v1.9/LanguageServer/jl_tF1RPU.
Stacktrace:
 [1] top-level scope
   @ ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/languageserver/main.jl:69
 [2] include(mod::Module, _path::String)
   @ Base ./Base.jl:418
 [3] exec_options(opts::Base.JLOptions)
   @ Base ./client.jl:299
 [4] _start()
   @ Base ./client.jl:516
[Info  - 5:12:17 PM] Connection to server got closed. Server will restart.
[ Info: Starting the Julia Language Server
ERROR: LoadError: ArgumentError: Package SparseArrays does not have Serialization in its dependencies:
- You may have a partially installed environment. Try `Pkg.instantiate()`
  to ensure all packages in the environment are installed.
- Or, if you have SparseArrays checked out for development and have
  added Serialization as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with SparseArrays
Stacktrace:
 [1] macro expansion
   @ ./loading.jl:1137 [inlined]
 [2] macro expansion
   @ ./lock.jl:267 [inlined]
 [3] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1114
 [4] include(mod::Module, _path::String)
   @ Base ./Base.jl:418
 [5] include(x::String)
   @ SparseArrays /nix/store/yddkhdhvkjq52b46x03gcndn7dg9d6b3-julia-latest-bin-latest/share/julia/stdlib/v1.9/SparseArrays/src/SparseArrays.jl:6
 [6] top-level scope
   @ /nix/store/yddkhdhvkjq52b46x03gcndn7dg9d6b3-julia-latest-bin-latest/share/julia/stdlib/v1.9/SparseArrays/src/SparseArrays.jl:63
 [7] include
   @ ./Base.jl:418 [inlined]
 [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
   @ Base ./loading.jl:1501
 [9] top-level scope
   @ stdin:1
in expression starting at /nix/store/yddkhdhvkjq52b46x03gcndn7dg9d6b3-julia-latest-bin-latest/share/julia/stdlib/v1.9/SparseArrays/src/solvers/umfpack.jl:3
in expression starting at /nix/store/yddkhdhvkjq52b46x03gcndn7dg9d6b3-julia-latest-bin-latest/share/julia/stdlib/v1.9/SparseArrays/src/SparseArrays.jl:3
in expression starting at stdin:1
ERROR: LoadError: Failed to precompile SparseArrays [2f01184e-e22b-5df5-ae63-d93ebab69eaf] to /home/u/.config/Code/User/globalStorage/julialang.language-julia/lsdepot/v1/compiled/v1.9/SparseArrays/jl_h1eMH0.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
    @ Base ./loading.jl:1652
  [3] compilecache
    @ ./loading.jl:1596 [inlined]
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1297
  [5] _require_prelocked(uuidkey::Base.PkgId)
    @ Base ./loading.jl:1170
  [6] macro expansion
    @ ./loading.jl:1150 [inlined]
  [7] macro expansion
    @ ./lock.jl:267 [inlined]
  [8] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1114
  [9] top-level scope
    @ ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/Compat/src/deprecated.jl:23
 [10] eval(m::Module, e::Any)
    @ Core ./boot.jl:370
 [11] top-level scope
    @ ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/Compat/src/deprecated.jl:22
 [12] include(mod::Module, _path::String)
    @ Base ./Base.jl:418
 [13] include(x::String)
    @ Compat ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/Compat/src/Compat.jl:1
 [14] top-level scope
    @ ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/Compat/src/Compat.jl:1328
 [15] include
    @ ./Base.jl:418 [inlined]
 [16] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1501
 [17] top-level scope
    @ stdin:1
in expression starting at /home/u/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/Compat/src/deprecated.jl:18
in expression starting at /home/u/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/Compat/src/Compat.jl:1
in expression starting at stdin:1
ERROR: LoadError: Failed to precompile Compat [34da2185-b29b-5c13-b0c7-acf172513d20] to /home/u/.config/Code/User/globalStorage/julialang.language-julia/lsdepot/v1/compiled/v1.9/Compat/jl_eC95r9.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
    @ Base ./loading.jl:1652
  [3] compilecache
    @ ./loading.jl:1596 [inlined]
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1297
  [5] _require_prelocked(uuidkey::Base.PkgId)
    @ Base ./loading.jl:1170
  [6] macro expansion
    @ ./loading.jl:1150 [inlined]
  [7] macro expansion
    @ ./lock.jl:267 [inlined]
  [8] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1114
  [9] include
    @ ./Base.jl:418 [inlined]
 [10] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1501
 [11] top-level scope
    @ stdin:1
in expression starting at /home/u/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/DataStructures/src/DataStructures.jl:1
in expression starting at stdin:1
ERROR: LoadError: Failed to precompile DataStructures [864edb3b-99cc-5e75-8d2d-829cb0a9cfe8] to /home/u/.config/Code/User/globalStorage/julialang.language-julia/lsdepot/v1/compiled/v1.9/DataStructures/jl_mMioie.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
    @ Base ./loading.jl:1652
  [3] compilecache
    @ ./loading.jl:1596 [inlined]
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1297
  [5] _require_prelocked(uuidkey::Base.PkgId)
    @ Base ./loading.jl:1170
  [6] macro expansion
    @ ./loading.jl:1150 [inlined]
  [7] macro expansion
    @ ./lock.jl:267 [inlined]
  [8] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1114
  [9] include
    @ ./Base.jl:418 [inlined]
 [10] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1501
 [11] top-level scope
    @ stdin:1
in expression starting at /home/u/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/JuliaFormatter/src/JuliaFormatter.jl:1
in expression starting at stdin:1
ERROR: LoadError: Failed to precompile JuliaFormatter [98e50ef6-434e-11e9-1051-2b60c6c9e899] to /home/u/.config/Code/User/globalStorage/julialang.language-julia/lsdepot/v1/compiled/v1.9/JuliaFormatter/jl_kVHhbr.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
    @ Base ./loading.jl:1652
  [3] compilecache
    @ ./loading.jl:1596 [inlined]
  [4] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1297
  [5] _require_prelocked(uuidkey::Base.PkgId)
    @ Base ./loading.jl:1170
  [6] macro expansion
    @ ./loading.jl:1150 [inlined]
  [7] macro expansion
    @ ./lock.jl:267 [inlined]
  [8] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1114
  [9] include
    @ ./Base.jl:418 [inlined]
 [10] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1501
 [11] top-level scope
    @ stdin:1
in expression starting at /home/u/.vscode/extensions/julialang.language-julia-1.6.24/scripts/packages/LanguageServer/src/LanguageServer.jl:1
in expression starting at stdin:1


The Language Server failed to precompile.
Please make sure you have permissions to write to the LS depot path at
	/home/u/.config/Code/User/globalStorage/julialang.language-julia/lsdepot/v1

┌ Error: Some Julia code in the VS Code extension crashed
└ @ Main ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/error_handler.jl:15
ERROR: Failed to precompile LanguageServer [2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7] to /home/u/.config/Code/User/globalStorage/julialang.language-julia/lsdepot/v1/compiled/v1.9/LanguageServer/jl_tl0Vom.
Stacktrace:
 [1] top-level scope
   @ ~/.vscode/extensions/julialang.language-julia-1.6.24/scripts/languageserver/main.jl:69
 [2] include(mod::Module, _path::String)
   @ Base ./Base.jl:418
 [3] exec_options(opts::Base.JLOptions)
   @ Base ./client.jl:299
 [4] _start()
   @ Base ./client.jl:516
Connection to server got closed. Server will not be restarted.
```

### PR DESCRIPTION
**Pull request by Liozou**
Fix #2939 (I hope).

To be frank, I don't have a clear view of the package management involved here, but manually tweaking the Manifest.toml like this PR does makes the extension work for me on master when following the workflow outlined at in [CONTRIBUTING.md](https://github.com/julia-vscode/julia-vscode/blob/master/CONTRIBUTING.md).

I think what caused #2939 might be related to the effort towards moving some stdlibs into external repository, (for example https://github.com/JuliaLang/julia/pull/44247, https://github.com/JuliaSparse/SparseArrays.jl/pull/95) but I could certainly be wrong.

### CODE REVIEW

**Comment by Liozou** [comment]
CI seems to indicate that this is only working starting starting from v1.7, because libblastrampoline_jll can not be installed on earlier versions of Julia, although the error message is a bit different depending on the versions. So this approach was too simplistic.

I'm happy to change the PR if anyone has a solution of course.

**Comment by pfitzseb** [comment]
This fix is correct, but we need to build the infrastructure for switching to a different env based on the Julia version.

**Comment by Liozou** [comment]
I tried to add an `environments-old` folder to separate environments between versions 1.7 and earlier, and 1.8 and later (see commit https://github.com/julia-vscode/julia-vscode/pull/2981/commits/49c14b42598706dfbc5068ed3d05de528202f3e5) but that seems to be failing, although I can't see the problem locally.

I also tried  to simply add SuiteSparse_jll and libblastrampoline_jll as new submodules (see commit https://github.com/julia-vscode/julia-vscode/pull/2981/commits/c7f733f51e6cc0efeab87dbe49e4a8fefd84ea8a) but that's not working either, though CI is green for julia 1.2, 1.3 and 1.7 curiously. So, another approach is needed.

File: .gitmodules [hunk 1]
```diff
@@ -76,3 +76,9 @@
 [submodule "scripts/packages/Revise"]
 	path = scripts/packages/Revise
 	url = https://github.com/timholy/Revise.jl
+[submodule "scripts/packages/libblastrampoline_jll"]
+	path = scripts/packages/libblastrampoline_jll
+	url = https://github.com/JuliaBinaryWrappers/libblastrampoline_jll.jl.git
+[submodule "scripts/packages/SuiteSparse_jll"]
+	path = scripts/packages/SuiteSparse_jll
+	url = https://github.com/JuliaBinaryWrappers/SuiteSparse_jll.jl.git
```
Comment (davidanthoff) [inline .gitmodules:84]: Why would we have to add more submodules?

**Comment by davidanthoff** [comment]
I might take a crack at a generic way to support Julia version specific environments. I'd almost prefer if we did this simply for each minor Julia version, rather than some hard split.

File: .gitmodules (hunk 1 above)
Comment (Liozou) [reply .gitmodules:84]: It was an attempt to solve the fact that older versions of Julia could not find them. But I believe the reason for that is that these packages actually do not have a version compatible with older julia versions, so the whole approach was doomed anyway.

**Comment by Liozou** [comment]
I'll close this PR since I don't believe this is the right way to handle this issue.

**Comment by davidanthoff** [comment]
See https://github.com/julia-vscode/julia-vscode/pull/2985.

**Pull request closed without merging**